File: stacks and queues/stack.js
class Node {
    constructor(value, next = null){
      this.value = value;
      this.next = next;
    }
  }
  
  class Stack {
    constructor(){
      this.top = null;
      this.bottom = null;
      this.length = 0;
    }
    peek() {
        //See 
        return this.top
    }
    push(value){
        const newNode = new Node(value)

        if (!this.length) {
          this.top = newNode
          this.bottom = newNode
        } else{
          const holding_top = this.top
          this.top = newNode
          this.top.next = holding_top
        }

        this.length++
        return this
    }
    pop(){
      if (this.top) {
        this.top = this.top.next
      } else{
        return
      }

      if(this.top === null) this.bottom = null
      this.length--
    }
    isEmpty(){
      if(this.length) return "Has values"

      return "Does not have values"
    }
  }
  
  const myStack = new Stack(); // Last in first out = LIFO

  console.log(myStack)
  myStack.push("google")
  myStack.push("Udemy")
  myStack.push("Discord")
  console.log(myStack)
  
  
  
  
  //Discord
  //Udemy
  //google

File: stacks and queues/stack.py
class Node:
    def __init__(self,data,next = None):
        self.data = data
        self.next = next

class Stack:
    def __init__(self) -> None:
        self.top = None
        self.bottom = None
        self.length = 0

    def peek(self):
        if self.top is None:
            print(f"Top is {self.top}")
        else:
             print(f"Value is {self.top}")

    

    def push(self,value):
        newNode = Node(value)

        if self.top == None:
            self.bottom = newNode
            self.top = newNode
        else:
            hold_top = self.top
            self.top = newNode
            self.top.next = hold_top
        
        self.length += 1
    
    def pop(self):
        if self.top == None:
            print("Empty")
            return
        elif self.top == self.bottom:
            self.top = None
            self.bottom = None
        else:
            self.top = self.top.next
            self.length -= 1
    
    def print_stack(self):
        if self.top == None:
            print("Stack empty")
        else:
            current_pointer = self.top
            while(current_pointer!=None):
                print(current_pointer.data)
                current_pointer = current_pointer.next
    

my_stack = Stack()

my_stack.peek()
my_stack.push("Google")
my_stack.push("Youtube")
my_stack.push("Discord")
my_stack.print_stack()